sunpy/data/_sample.py

```diff
@@ -23,93 +22,118 @@
         The timeout in seconds. If `None` the default timeout is used from
         `astropy.utils.data.Conf.remote_timeout`.
 
     Returns
     -------
-    None
+    result: `str`
+        The local path of the file. None if it failed.
     """
-    # Creating the directory for sample files to be downloaded
-    sampledata_dir = get_and_create_sample_dir()
-
-    number_of_files_fetched = 0
-    print("Downloading sample files to {}".format(sampledata_dir))
-    for file_name in six.itervalues(_files):
-        if not overwrite:
-            if os.path.isfile(os.path.join(sampledata_dir,
-                                           file_name[0])):
-                number_of_files_fetched += 1
-                continue
-
-        for base_url in _base_urls:
-            full_file_name = file_name[0] + file_name[1]
+
+    if filename[-3:] == 'zip':
+        uncompressed_filename = filename[:-4]
+    else:
+        uncompressed_filename = filename
+    # check if the (uncompressed) file exists
+    if not overwrite and os.path.isfile(os.path.join(sampledata_dir, uncompressed_filename)):
+        return os.path.join(sampledata_dir, uncompressed_filename)
+    else:
+        # check each provided url to find the file
+        for base_url in url_list:
+            online_filename = filename
+            if base_url.count('github'):
+                online_filename += '?raw=true'
             try:
-                exists = url_exists(os.path.join(base_url, full_file_name))
+                exists = url_exists(os.path.join(base_url, online_filename))
                 if exists:
-                    f = download_file(os.path.join(base_url, full_file_name))
-                    real_name, ext = os.path.splitext(full_file_name)
+                    f = download_file(os.path.join(base_url, online_filename), show_progress=show_progress)
+                    real_name, ext = os.path.splitext(f)
 
-                    if file_name[1] == '.zip':
+                    if ext == '.zip':
                         print("Unpacking: {}".format(real_name))
                         with ZipFile(f, 'r') as zip_file:
-                            zip_file.extract(real_name, sampledata_dir)
+                            unzipped_f = zip_file.extract(real_name, sampledata_dir)
                         os.remove(f)
+                        move(unzipped_f, os.path.join(sampledata_dir, uncompressed_filename))
+                        return os.path.join(sampledata_dir, uncompressed_filename)
                     else:
                         # move files to the data directory
-                        move(f, os.path.join(sampledata_dir, file_name[0]))
-                    # increment the number of files obtained to check later
-                    number_of_files_fetched += 1
-                    break
+                        move(f, os.path.join(sampledata_dir, uncompressed_filename))
+                        return os.path.join(sampledata_dir, uncompressed_filename)
```

Comment:
it doesn't actually repeat as it only get hits once a single file gets downloaded